sklearn/tests/test_kernel_approximation.py

```diff
@@ -114,34 +114,41 @@ def test_additive_chi2_sampler():
     Y_neg[0, 0] = -1
     msg = "Negative values in data passed to"
     with pytest.raises(ValueError, match=msg):
-        transform.transform(Y_neg)
+        transformer.fit(Y_neg)
+
+    # test that the sample interval is set correctly
+    sample_intervals = [0.8, 0.5, 0.4]
+    for sample_interval in sample_intervals:
+        # test that the sample_interval is initialized correctly
+        transformer = AdditiveChi2Sampler(sample_interval=sample_interval)
+        transformer.fit(X)
```

Comment:
I think this is the part where we need to call both `fit_transform` and `transform` with the different possible `sample_intervals`.

Do no hesitate to isolate this inside a test to be able to parametrize on the method and value of `sample_intervals` easily.